### PR TITLE
Feat/met 2792 create depublish workflow

### PR DESCRIPTION
**Pull request by stzanakis**

### CODE REVIEW

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractExecutablePlugin.java [hunk 1]
```diff
@@ -171,6 +171,15 @@ DpsTask createDpsTaskForIndexPlugin(EcloudBasePluginParameters ecloudBasePluginP
     return createDpsTaskForProcessPlugin(ecloudBasePluginParameters, extraParameters);
   }
 
+  DpsTask createDpsTaskForDepublishPlugin(EcloudBasePluginParameters ecloudBasePluginParameters,
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractExecutablePlugin.java:174]: The first parameter is never used. I guess you forgot to apply the base parameters?

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package eu.europeana.metis.core.workflow.plugins;
+
+import eu.europeana.cloud.service.dps.DpsTask;
+import java.util.List;
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java:4]: Unnecessary import.

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  /**
+   * Constructor to initialize the plugin with pluginMetadata.
+   * <p>Initializes the {@link #pluginType} as well.</p>
+   *
+   * @param pluginMetadata The plugin metadata.
+   */
+  DepublishPlugin(DepublishPluginMetadata pluginMetadata) {
+    super(PluginType.PUBLISH, pluginMetadata);
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java:32]: Wrong PluginType choice?

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import eu.europeana.cloud.service.dps.DpsTask;
+import java.util.List;
+
+/**
+ * Depublish Plugin.
+ *
+ * @author Simon Tzanakis (Simon.Tzanakis@europeana.eu)
+ * @since 2020-06-16
+ */
+public class DepublishPlugin extends AbstractExecutablePlugin<DepublishPluginMetadata> {
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java:12]: Knowing that there will be a depublish plugin for records later, perhaps rename this DepublishDatasetPlugin? Or would you prefer to merge these two plugins?

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPluginMetadata.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package eu.europeana.metis.core.workflow.plugins;
+
+import java.util.ArrayList;
+import java.util.List;
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPluginMetadata.java:4]: Two unused imports... :)

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java (hunk 4 above)
Comment (jochen-vermeulen) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java:12]: In fact, this goes for all enum members that you declared as well (e.g. those called DEPUBLISH would be called DEPUBLISH_DATASET).

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java (hunk 3 above)
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java]: Indeed :D

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java (hunk 4 above)
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java:11]: Actually I was thinking indeed to combine the two in one plugin indeed and we'll have two separate parameters. What do you think?

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java (hunk 2 above)
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPlugin.java]: Weird my ide didn't handle it automatically

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPluginMetadata.java (hunk 5 above)
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/DepublishPluginMetadata.java]: Something has seriously gone wrong with my ide, the new update I guess, darn...

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractExecutablePlugin.java (hunk 1 above)
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractExecutablePlugin.java]: No actually they are not needed.. So I don't need the parameter here..

**Comment by stzanakis** [comment]
Closing this, PR #350 is the correct one.

**Pull request closed without merging**